kafka-ui-react-app/src/components/Brokers/Broker/Broker.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import { ClusterName } from 'redux/interfaces';
+import useInterval from 'lib/hooks/useInterval';
+import { useParams } from 'react-router-dom';
+import PageHeading from 'components/common/PageHeading/PageHeading';
+import { BrokersApi, Configuration } from 'generated-sources';
+import { BASE_PARAMS } from 'lib/constants';
+import { Table } from 'components/common/table/Table/Table.styled';
+import TableHeaderCell from 'components/common/table/TableHeaderCell/TableHeaderCell';
+import * as Metrics from 'components/common/Metrics';
+import { useAppDispatch, useAppSelector } from 'lib/hooks/redux';
+import {
+  fetchBrokers,
+  fetchClusterStats,
+  selectStats,
+} from 'redux/reducers/brokers/brokersSlice';
+import BytesFormatted from 'components/common/BytesFormatted/BytesFormatted';
+
+const apiClientConf = new Configuration(BASE_PARAMS);
+export const brokersApiClient = new BrokersApi(apiClientConf);
+
+interface BrokerLogdirState {
+  name: string;
+  error: string;
+  topics: number;
+  partitions: number;
+}
+
+const Broker: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { clusterName, brokerId } =
+    useParams<{ clusterName: ClusterName; brokerId: string }>();
+
+  const [logdirs, setLogdirs] = useState<BrokerLogdirState>();
+  const { diskUsage, items } = useAppSelector(selectStats);
+
+  const fetchData = async () => {
+    const res = await brokersApiClient.getAllBrokersLogdirs({
+      clusterName: clusterName as string,
+      broker: [Number(brokerId)],
+    });
+    if (res && res[0]) {
+      const partitionsCount =
+        res[0].topics?.reduce(
+          (prevValue, value) => prevValue + (value.partitions?.length || 0),
+          0
+        ) || 0;
+
+      const brokerLogdir = {
+        name: res[0].name || '-',
+        error: res[0].error || '-',
+        topics: res[0].topics?.length || 0,
+        partitions: partitionsCount,
+      };
+      setLogdirs(brokerLogdir);
+    }
+
+    dispatch(fetchClusterStats(clusterName));
+    dispatch(fetchBrokers(clusterName));
+  };
+
+  React.useEffect(() => {
+    fetchData().then();
+  }, [clusterName, brokerId, dispatch]);
+
+  const brokerItem = items?.find(
+    (item) => Number(item.id) === Number(brokerId)
+  );
+  const brokerDiskUsage = diskUsage?.find(
+    (item) => Number(item.brokerId) === Number(brokerId)
```

Comment:
Why we need to cast both? What the initial type for `item.brokerId` & `brokerId`?